src/clone.rs

```diff
@@ -81,45 +91,52 @@ pub fn clone(args: InputArgs, mut clone_repos: Vec<GitRepo>) {
         println!("{}", "Please configure conf file to clone repositories or pass the necessary values as arguments".blue())
     }
 
-    for remote in remotes_from_args {
-        let mut use_ssh = false;
-        if remote.remote_url.contains("git@") {
-            use_ssh = true;
-        }
-
-        let mut clone = GitClone {
-            remote_url: remote.remote_url.as_str(),
-            local_path: Path::new(remote.local_path.as_str()),
-            use_ssh: use_ssh,
-        };
-        clone.git_action().expect(format!("Failed to clone repo {}, ", remote.remote_url).as_str());
-    }
+    let multi_bars = ProgressTracker::new();
+    remotes_from_args
+        .into_iter()
+        .map(|remote| GitClone {
+            use_ssh: remote.remote_url.contains("git@"),
+            remote_url: remote.remote_url,
+            local_path: remote.local_path.into(),
+        })
+        .for_each(|clone| multi_bars.start_task(clone));
+
```

Comment:
There seems to be a time period where no progress is shown before the download actually begins. Only the remote is printed. Can we show some somethign that indicates a download is about to begin??

Reply:
By default it should print Waiting for process to begin, we might print the remote message (things like
```
remote: Enumerating objects: 6915, done.
remote: Counting objects: 100% (4012/4012), done.
remote: Compressing objects: 100% (2547/2547), done.
remote: Total 2966 (delta 2589), reused 487 (delta 414)
```
), but in my experience, it was only disturbing the screen

Reply:
Yeah, dont print that. just a line saying cloning has begun. The clone command already has this, but its outside ur progress bar which is fine. For create, there is no feedback as to what is happening. After remote creation, can we just print that cloning will begin.

Reply:
Except for the bug in single threaded mode, it should now print `Waiting for process to begin` while waiting for a free thread, and `Process is about to begin` before the download really start